### ISSUE TITLE
catch-or-return has a false-positive when used with .finally(..)

### ISSUE THREAD

**Issue by evancharlton** [yannickcr/eslint-plugin-react#2650]
Hi there!

I have this interesting situation in my app. Here's the code that made me raise the issue:
```js
  axios
    .get('/')
    .then((a) => {
      console.log(a);
    });
```
Triggers an error: `error  Expected catch() or return  promise/catch-or-return`
Great, lets fix that:
```js
  axios
    .get('/')
    .then((a) => {
      console.log(a);
    })
    .catch((ex) => {
      console.error(ex);
    });
```
Yay, everything is happy! But now let's add something in there at the end:
```js
  axios
    .get('/')
    .then((a) => {
      console.log(a);
    })
    .catch((ex) => {
      console.error(ex);
    })
    .finally(() => {
      console.log('All done');
    });
```
I now have another error: `error    Expected catch() or return    promise/catch-or-return`

What did I screw up? Here's the eslint-related stuff in my `package.json`:
```json
    "eslint": "^6.8.0",
    "eslint-config-airbnb": "^18.1.0",
    "eslint-config-prettier": "^6.11.0",
    "eslint-plugin-import": "^2.20.2",
    "eslint-plugin-jest": "^23.13.1",
    "eslint-plugin-lodash": "^7.1.0",
    "eslint-plugin-prettier": "^3.1.3",
    "eslint-plugin-promise": "^4.2.1",
    "eslint-plugin-testcafe": "^0.2.1",
    "eslint-plugin-testing-library": "^3.1.3",
```

**Comment by evancharlton** [comment]
Note that if I assign it to a variable, all of the conditions are happy.
```js
  const foo = axios
    .get('/')
    .then((a) => {
      console.log(a);
    })
    .catch((ex) => {
      console.error(ex);
    })
    .finally(() => {
      console.log('All done');
    });
```

**Comment by ljharb** [comment]
This rule is from eslint-plugin-promise, which isn’t this repo. Please file it there.

**Comment by evancharlton** [comment]
Oh lordy, I knew that -- but I have no idea how I wound up in this repo. I'm sorry!
